### PR TITLE
Cleanup and enhance pull_request_template.

### PR DESCRIPTION
**Pull request by bendory**
# Changes

- Move all link definitions down to the bottom for readability.
- Added request for `kind` label
- Added link to end-to-end testing documentation

/kind documentation

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [ ] ~Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)~ **N/A**
- [ ] ~Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)~ **N/A**
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [ ] ~Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]~ **N/A**

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

**Comment by bendory** [comment]
/assign @chmouel 

/assign @imjasonh 

I got assigned to 2 people on parental leave. :-(

File: .github/pull_request_template.md
Comment (vinamra28) [inline .github/pull_request_template.md:51]: I guess you forgot append `:` 🙂

Comment (bendory) [reply .github/pull_request_template.md]: Good catch, thanks! Fixed.

**Comment by afrittoli** [comment]
Looks good to me, but I'll leave it to the @tektoncd/catalog-maintainers to approve.

**Review by vinamra28 — APPROVED** [review]

**Comment by vinamra28** [comment]
@bendory, can you please rebase your PR over upstream main branch? (This will unblock your PR)

**Review by jerop — COMMENTED** [review]
/lgtm

**Pull request merged**
